### PR TITLE
Add API to read and write ICEkit Image data

### PR DESCRIPTION
**Pull request by jmurty**
Add API endpoints to list, edit, and add data for ICEkit's Image model.

You can now access the images API at the web UI at _/api/images/_ when you are logged in as a user with sufficient permissions (either a super-user, or a user with change/add/delete model permissions for Image)

Use the new tokens admin at _/admin/authtoken/token/_ to generate opaque per-user authentication tokens, then include this token value in requests from clients outside a web browser context (i.e. which cannot or should not use standard session authentication)

Example curl commands to perform API operations with token authorisation (and assuming the token's associated user has sufficient permissions):

* List Images:  
  `curl -X GET -H 'Authorization: Token 6c84357926bb0e3cb359e4b2e89ee7fa7c142731' http://icekit.lvh.me:8000/api/images/`
* Get details of a specific image PK=1:  
  `curl -X GET -H 'Authorization: Token 6c84357926bb0e3cb359e4b2e89ee7fa7c142731' http://icekit.lvh.me:8000/api/images/1/`
* Create an Image:  
  `curl -X POST -H 'Authorization: Token 6c84357926bb0e3cb359e4b2e89ee7fa7c142731' --form title='New image' --form image=@image_file.jpg http://icekit.lvh.me:8000/api/images/`
* Replace an image (**must** include image file data):  
  `curl -X PUT -H 'Authorization: Token 6c84357926bb0e3cb359e4b2e89ee7fa7c142731' --form title='Replaced image' --form image=@image_file.jpg http://icekit.lvh.me:8000/api/images/1/`
* Update an image (can be with partial changes, image file data need not be included):  
  `curl -X PATCH -H 'Authorization: Token 6c84357926bb0e3cb359e4b2e89ee7fa7c142731' --form title='Updated image' http://icekit.lvh.me:8000/api/images/1/`
* Delete an image:  
  `curl -X DELETE -H 'Authorization: Token 6c84357926bb0e3cb359e4b2e89ee7fa7c142731' http://icekit.lvh.me:8000/api/images/1/`

Side-effects on GLAMkit as a result of this work include:
* include Django REST framework in installed apps by default
* lock down authorisation and permission settings for DRF APIs in general (the existing pages API is explicitly re-permitted for anyone)
* enable Token based authentication for access to APIs by default.

### CODE REVIEW

File: icekit/images_api/tests.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        self.assertEqual(403, response.status_code)
+
+        response = self.client.delete(
+            reverse('images-detail', args=[self.image.pk]),
+            HTTP_AUTHORIZATION='Token %s' % self.user_token.key)
+        self.assertEqual(403, response.status_code)
+
+        # User with 'change' model permissions can list and change existing
+        # but not add or delete
+        self.user.user_permissions = [
```
Comment (cogat) [inline icekit/images_api/tests.py]: Hrm, I think we need a permission to allow read access independently of change access, and independently of possessing a token - some API consumers may need read permission but not write permission. Or else a specific `can access image API` permission that is necessary to access any part of API and used in combination with django's built-in ones.

We may introduce a public-level permission in future. Whatever we decide should work similarly on ICEkit's other APIs, and with a 'public' addition in future.

File: icekit/images_api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from django.apps import apps
+
+from rest_framework import serializers
+
+
+Image = apps.get_model('icekit_plugins_image.Image')
```
Comment (cogat) [inline icekit/images_api/serializers.py]: Could you provide a touch of documentation in `docs` and `changelog`? I suggest it doesn't need much more than starting an `topics/apis.rst` doc and copying over the PR comment.

File: icekit/project/urls.py [hunk 3]
```diff
@@ -40,8 +40,11 @@
         {'sitemaps': sitemaps}),
 
     # Installed apps.
-    url(r'^api/pages/', include('icekit.pages_api.urls')),
     url(r'^forms/', include('forms_builder.forms.urls')),
+    # APIs
+    # TODO This URL routing technique doesn't show all endpoints in root /api/
+    url(r'^api/', include('icekit.pages_api.urls')),
```
Comment (cogat) [inline icekit/project/urls.py]: Worth addressing now I think. I'd suggest doing the router registration here, not in each separate `urls.py`

File: icekit/images_api/tests.py (hunk 1 above)
Comment (jmurty) [inline icekit/images_api/tests.py]: The DRF handling of permissions and authorisation is pretty flexible, and defaults can be overridden with the `permission_classes` and `authentication_classes` decorators on a per-view or per-viewset basis.

Ultimately we will probably need a way to configure API permission/authentication settings based on environment variables, but we can leave it until we really need this.

File: icekit/project/urls.py (hunk 3 above)
Comment (jmurty) [inline icekit/project/urls.py]: Top-level URL routing will be fixed shortly by the general app reorganisation

**Pull request merged**